File: sound-parent/sound-api/src/main/java/com/wisdom/sound/service/GroupService.java
package com.wisdom.sound.service;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import com.wisdom.sound.entity.PageResult;
import com.wisdom.sound.pojo.Group;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface GroupService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<Group> findAll();

	public List<HashMap<String,String>> selectForSelect2();
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(Group group);
	
	
	/**
	 * 修改
	 */
	public void update(Group group);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public Group findOne(String id);


	public Group test();
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(String[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(Group group, int pageNum, int pageSize);
	
}

File: sound-parent/sound-web/src/main/resources/static/js/service/groupService.js
//服务层
app.service('groupService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('/group/findAll');
	}

	//读取列表数据绑定到表单中
	this.selectForSelect2=function(){
		return $http.get('/group/groupList');
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('/group/findPage?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('/group/findOne?groupId='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('/group/add',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('/group/update',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('/group/delete?ids='+ids);
	}
	//搜索
	 this.search=function(page,rows,searchEntity){
		return $http.post('/group/search?page='+page+"&rows="+rows, searchEntity);
	 }
});
